cli/src/QueryNodeApi.ts

```diff
@@ -156,18 +171,35 @@ export default class QueryNodeApi {
   }
 
   async storageBucketsForNewChannel(): Promise<StorageNodeInfoFragment[]> {
-    return this.multipleEntitiesQuery<GetStorageBucketsQuery, GetStorageBucketsQueryVariables>(
-      GetStorageBuckets,
+    const countQueryResult = await this.uniqueEntityQuery<StorageBucketsCountQuery, StorageBucketsCountQueryVariables>(
+      StorageBucketsCount,
       {},
+      'storageBucketsConnection'
+    )
+    if (!countQueryResult) {
+      throw Error('Invalid query. Could not fetch storage buckets count information')
+    }
+
+    const buckets = await this.multipleEntitiesQuery<GetStorageBucketsQuery, GetStorageBucketsQueryVariables>(
+      GetStorageBuckets,
+      { count: countQueryResult.totalCount },
       'storageBuckets'
     )
+
+    // sorting buckets based on available size, if two buckets have same
+    // available size then sort the two based on available dataObjects count
+    return buckets.sort(
+      (x, y) =>
+        y.dataObjectsSizeLimit - y.dataObjectsSize - (x.dataObjectsSizeLimit - x.dataObjectsSize) ||
+        y.dataObjectCountLimit - y.dataObjectsCount - (x.dataObjectCountLimit - x.dataObjectsCount)
```

Comment:
`dataObjectsSize`, `dataObjectsSizeLimit`, `dataObjectsCount` and  `dataObjectCountLimit` have `BigNumber` type in the QN, so the api returns them as `string`. Using arithmetic operators on strings may give some unexpected results, you should convert them to `BN`s first, ie:
```suggestion
        new BN(y.dataObjectsSizeLimit)
          .sub(new BN(y.dataObjectsSize))
          .cmp(new BN(x.dataObjectsSizeLimit).sub(new BN(x.dataObjectsSize))) ||
        new BN(y.dataObjectCountLimit)
          .sub(y.dataObjectsCount)
          .cmp(new BN(x.dataObjectCountLimit).sub(new BN(x.dataObjectsCount)))
```